sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ThroughputControlStore.java

```diff
@@ -151,38 +154,53 @@ public void enableThroughputControlGroup(ThroughputControlGroup group) {
         return this.resolveContainerController(collectionLink)
             .flatMap(containerController -> {
                 if (containerController.canHandleRequest(request)) {
-                    return Mono.just(containerController);
+                    return containerController.processRequest(request, originalRequestMono)
+                        .doOnError(throwable -> this.handleException(collectionLink, request, throwable));
                 }
 
                 // Unable to find container controller to handle the request,
                 // It is caused by control store out of sync or the request has staled info.
                 // We will handle the first scenario by creating a new container controller,
                 // while fall back to original request Mono for the second scenario.
-                return this.shouldRefreshContainerController(collectionLink, request)
-                    .flatMap(shouldRefresh -> {
-                        if (shouldRefresh) {
-                            containerController.close().subscribeOn(Schedulers.parallel()).subscribe();
-                            this.containerControllerCache.refresh(collectionLink, () -> this.createAndInitContainerController(collectionLink));
-                            return this.resolveContainerController(collectionLink);
+                return this.updateControllerAndRetry(collectionLink, request, originalRequestMono);
+            });
+    }
+
+    private <T> Mono<T> updateControllerAndRetry(
+        String collectionLink,
```

Comment:
is this name link or self link?
Could you please add a comment or change arg name to be clear (.e.g, collectionSelfLink or collectionNameLink)

Reply:
This is name link. Will change the name to be more clear.

Reply:
Updated.